packages/core/src/reducers/activities.ts

```diff
@@ -134,33 +138,91 @@ export default function activities(
           payload: { activity }
         } = action;
 
+        // `channelData.state` is being deprecated in favor of `channelData['webchat:send-status']`.
+        // Please refer to #4362 for details. Remove on or after 2024-07-31.
         activity = updateIn(activity, ['channelData', 'state'], () => SENDING);
+        activity = updateIn(activity, ['channelData', 'webchat:send-status'], () => 'sending');
 
         state = upsertActivityWithSort(state, activity);
       }
 
       break;
 
-    case POST_ACTIVITY_REJECTED:
+    case POST_ACTIVITY_IMPEDED:
```

Comment:
"Impeded" = the activity past the `styleOptions.sendTimeout` period.

This is only needed for backward compatible with previous `channelData.state` field and should be removed after deprecation period.